### PR TITLE
Fix prompt while spinning

### PR DESCRIPTION
**Pull request by vhvb1989**
- remove call to asyncContext
- make console aware of interaction

fix: https://github.com/Azure/azure-dev/issues/1335

### CODE REVIEW

File: cli/azd/pkg/input/console.go [hunk 1]
```diff
@@ -345,3 +350,23 @@ func GetStepResultFormat(result error) SpinnerUxType {
 	}
 	return formatResult
 }
+
+// Handle doing interactive calls. It check if there's a spinner running to pause it before doing interactive actions.
+func (c *AskerConsole) doInteraction(fn func(c *AskerConsole) error) error {
+	var resumeSpinner bool
+	if c.spinner != nil && c.spinner.Status() == yacspin.SpinnerRunning {
+		_ = c.spinner.Pause()
+		resumeSpinner = true
```
Comment (ellismg) [inline cli/azd/pkg/input/console.go:359]: I am wondering if it is better to instead do:

```go
defer func() { _ = c.spinner.Unpasuse() }()
```

As we have it now, if the worker function errors, the spinner will remain paused.  If something upstack "recovered" from the error, the UI would be in an inconsistent state.  Using defer here would ensure that we resume the spinner are all exit paths.

**Review by weikanglim — APPROVED** [review]
LGTM

**Review by ellismg — APPROVED** [review]

**Comment by vhvb1989** [comment]
/check-enforcer override
since changes are not related to the failing gate for verifying installers

**Pull request merged**